### ISSUE TITLE
[Bug]: Notifications API returns 500 error

### ISSUE THREAD

**Issue by melroy89** [nextcloud/notifications#2069]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

The official OCS Notification API v2 is running 500 server error. More details at the bottom!

(I already try to logout and login again, but the problem persists).

https://cloud.melroy.org/ocs/v2.php/apps/notifications/api/v2/notifications

![image](https://github.com/user-attachments/assets/65cc1a87-3f72-4ee0-9dfd-c956ceb87e84)

---

Other bug: I also can't select Nextcloud server 30 from the drop-down menu in this issue template of GitHub... During creating this issue... PR created for this: https://github.com/nextcloud/server/pull/47973

### Steps to reproduce

1. Upgrade to Nextcloud v30 (Hub 9)
2. Open the Developer tools (eg. in Firefox)
3. Go to the Nextcloud homepage of the server
4. Notice a 500 error on the notifcations request (I also get errors on the Android app under the notification tab)

### Expected behavior

No 500 server error on Notifications API v2.

### Nextcloud Server version

30

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.3

### Web server

Nginx

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Upgraded to a MAJOR version (ex. 28 to 29)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "cloud.melroy.org"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "default_timezone": "Europe\/Amsterdam",
        "overwrite.cli.url": "https:\/\/cloud.melroy.org\/",
        "dbtype": "mysql",
        "filelocking.enabled": "true",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0,
            "dbindex": 0,
            "timeout": 1.5
        },
        "default_language": "nl",
        "default_locale": "nl_NL",
        "default_phone_region": "NL",
        "version": "30.0.0.14",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "theme": "",
        "maintenance": false,
        "loglevel": 3,
        "trashbin_retention_obligation": "auto",
        "updater.release.channel": "stable",
        "mail_smtpsecure": "ssl",
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpauth": 1,
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "mysql.utf8mb4": true,
        "app_install_overwrite": [
            "mail",
            "files_markdown",
            "news",
            "files_rightclick",
            "files_antivirus",
            "checksum",
            "richdocuments"
        ],
        "maintenance_window_start": 2,
        "defaultapp": "files"
    }
}
```

### List of activated Apps

```shell
Enabled:
  - activity: 3.0.0
  - admin_audit: 1.20.0
  - bruteforcesettings: 3.0.0
  - calendar: 5.0.0
  - cloud_federation_api: 1.13.0
  - comments: 1.20.1
  - contacts: 6.1.0
  - contactsinteraction: 1.11.0
  - dav: 1.31.1
  - federatedfilesharing: 1.20.0
  - federation: 1.20.0
  - files: 2.2.0
  - files_downloadlimit: 3.0.0
  - files_markdown: 2.4.1
  - files_reminders: 1.3.0
  - files_sharing: 1.22.0
  - files_trashbin: 1.20.1
  - files_versions: 1.23.0
  - firstrunwizard: 3.0.0
  - forms: 4.3.0-alpha7
  - logreader: 3.0.0
  - lookup_server_connector: 1.18.0
  - news: 25.0.0-alpha8
  - nextcloud_announcements: 2.0.0
  - notes: 4.10.1
  - notifications: 3.0.0
  - oauth2: 1.18.1
  - password_policy: 2.0.0
  - photos: 3.0.2
  - privacy: 2.0.0
  - provisioning_api: 1.20.0
  - recommendations: 3.0.0
  - related_resources: 1.5.0
  - serverinfo: 2.0.0
  - settings: 1.13.0
  - sharebymail: 1.20.0
  - spreed: 20.0.0
  - survey_client: 2.0.0
  - systemtags: 1.20.0
  - tasks: 0.16.1
  - text: 4.1.0
  - theming: 2.5.0
  - twofactor_backupcodes: 1.19.0
  - twofactor_totp: 12.0.0-dev
  - updatenotification: 1.20.0
  - user_status: 1.10.0
  - viewer: 3.0.0
  - webhook_listeners: 1.1.0-dev
  - workflowengine: 2.12.0
Disabled:
  - checksum: 1.2.4 (installed 1.2.4)
  - circles: 30.0.0-dev (installed 28.0.0-dev)
  - dashboard: 7.10.0 (installed 7.0.0)
  - encryption: 2.18.0 (installed 2.0.0)
  - files_antivirus: 5.5.7 (installed 5.5.7)
  - files_external: 1.22.0 (installed 1.20.0)
  - files_pdfviewer: 3.0.0 (installed 2.0.1)
  - mail: 4.0.0 (installed 4.0.0)
  - richdocuments: 8.5.0 (installed 8.5.0)
  - richdocumentscode: 24.4.702 (installed 24.4.702)
  - support: 2.0.0 (installed 1.11.0)
  - suspicious_login: 8.0.0
  - twofactor_nextcloud_notification: 4.0.0
  - user_ldap: 1.21.0
  - weather_status: 1.10.0 (installed 1.8.0)
```

### Nextcloud Signing status

```shell
Path or page not found?? This still URL still valid for Nextcloud v30?
```

If you would mean `php ./occ integrity:check-core`? Then this command run fine, without any output.

### Nextcloud Logs

Note: When I get the 500 error, I don't see any error in the error log anymore...  So `"OCP\\HintException: Configuration was not read or initialized correctly, not overwriting /var/www/nextcloud/html/config/config.php at /var/www/nextcloud/html/lib/private/Config.php#266` error message only happened once!?

```shell
{"reqId":"LolyJrWAoDPenXefXoT8","level":3,"time":"2024-02-17T00:43:16+00:00","remoteAddr":"192.168.1.1","user":"--","app":"PHP","method":"GET","url":"/ocs/v2.php/apps/notifications/api/v2/notifications","message":"OCP\\HintException: Configuration was not read or initialized correctly, not overwriting /var/www/nextcloud/html/config/config.php at /var/www/nextcloud/html/lib/private/Config.php#266","userAgent":"Mozilla/5.0 (X11; Linux x86_64; rv:122.0) Gecko/20100101 Firefox/122.0","version":"","data":{"app":"PHP"}}
{"reqId":"K1PCzHpoIYMi96Z145rU","level":3,"time":"2024-02-17T00:43:30+00:00","remoteAddr":"192.168.1.1","user":"--","app":"PHP","method":"PROPFIND","url":"/remote.php/dav/files/melroy/","message":"Error: Class \"OCA\\DAV\\Connector\\Sabre\\ExceptionLoggerPlugin\" not found at /var/www/nextcloud/html/remote.php#62","userAgent":"Mozilla/5.0 (Linux) mirall/3.12.0 (build 20570) (Nextcloud, linuxmint-6.5.0-17-generic ClientArchitecture: x86_64 OsArchitecture: x86_64)","version":"","data":{"app":"PHP"}}
```

### Additional info

```sh
Request URL: https://cloud.melroy.org/ocs/v2.php/apps/notifications/api/v2/notifications
```
---

Request Headers: 

```sh
GET /ocs/v2.php/apps/notifications/api/v2/notifications HTTP/3
Host: cloud.melroy.org
User-Agent: Mozilla/5.0 (X11; Linux x86_64; rv:128.0) Gecko/20100101 Firefox/128.0
Accept: application/json, text/plain, */*
Accept-Language: en-US,en;q=0.5
Accept-Encoding: gzip, deflate, br, zstd
X-Requested-With: XMLHttpRequest, XMLHttpRequest
requesttoken:secret
Sec-Fetch-Dest: empty
Sec-Fetch-Mode: cors
Sec-Fetch-Site: same-origin
Connection: keep-alive
Alt-Used: cloud.melroy.org
Cookie: __Host-nc_sameSiteCookielax=true; __Host-nc_sameSiteCookiestrict=true; oc_sessionPassphrase=secret; nc_username=melroy; nc_token=secret; nc_session_id=secret
Pragma: no-cache
Cache-Control: no-cache
TE: trailers
```
Response:

---

```xml
<?xml version="1.0"?>
<ocs>
 <meta>
  <status>failure</status>
  <statuscode>500</statuscode>
  <message>Internal Server Error
</message>
 </meta>
 <data/>
</ocs>

```

Response headers:

```sh
HTTP/3 500 
server: nginx
date: Sat, 14 Sep 2024 13:13:58 GMT
content-type: text/xml; charset=UTF-8
content-security-policy: default-src 'self'; script-src 'self' 'secret'; style-src 'self' 'unsafe-inline'; frame-src *; img-src * data: blob:; font-src 'self' data:; media-src *; connect-src *; object-src 'none'; base-uri 'self';
alt-svc: h3=":443";ma=86400,h3-29=":443";ma=86400,h3-27=":443";ma=86400
x-quic: h3
x-robots-tag: noindex, nofollow
x-frame-options: SAMEORIGIN
x-xss-protection: 1; mode=block
x-content-type-options: nosniff
referrer-policy: no-referrer
x-download-options: noopen
x-permitted-cross-domain-policies: none
strict-transport-security: max-age=31536000; includeSubDomains; preload
```

Nginx log message: `192.168.1.102 - - [14/Sep/2024:23:02:36 +0200]  "GET /ocs/v2.php/apps/notifications/api/v2/notifications HTTP/3.0" 500 166 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:128.0) Gecko/20100101 Firefox/128.0" "cloud.melroy.org" 0.017`

I already tried to enable loglevel 2 or 1, but no errors in the log.. I reload PHP8.3-fpm service as well to avoid having cache issues. Nothing works 😢 . And I do not know how to debug the 500 error, apart from saying `Internal Server Error`.

**Comment by nickvergessen** [comment]
> This may sound stupid or unrelated, but I had the same issue and noticed a lot of errors in the protocol as described in this Thread.

- It's not stupid and not unrelated. The firstrunwizard promotes some apps via notifications for admin.
- If the app is installed this specific notification is removed.
- For 30 the whiteboard was added, so installing it removes the broken notification.
- Unluckily the icon was forgotten to be added
- It was fixed for the upcoming 30.0.1 with https://github.com/nextcloud/firstrunwizard/pull/1363
- another thing that could magically make it work and break again is when you receive 25+ newer notifications (fixed) and afterwards resolve some again (broken again)

TLDR: 
- copy icon from https://github.com/nextcloud/firstrunwizard/pull/1363 into apps/firstrunwizard directory
- or install whiteboard app

**Comment by joshtrichards** [comment]
v30.0.1 has since been published. Closing.

**Comment by melroy89** [comment]
> v30.0.1 has since been published. Closing.

You're correct, I can confirm, with v30.0.1 I don't see this error anymore. Thanks!
